### PR TITLE
Try not to take down half a cluster if we dont need to

### PR DESCRIPTION
**Pull request by paololim**

### CODE REVIEW

**Comment by ctumolosus** [comment]
OMG Yes!

File: api/app/aws/EC2ContainerService.scala [hunk 1]
```diff
@@ -438,7 +438,13 @@ case class EC2ContainerService @javax.inject.Inject() (
       val minimumHealthyPercent = if (cfg.allowDowntime.getOrElse(false)) {
         0
       } else {
-        50
+        if (serviceDesiredCount >= 4) {
+          75 // take down one at a time, keep 3+ rurnning
```
Comment (chrhicks) [inline api/app/aws/EC2ContainerService.scala]: could make this `1/serviceDesiredCount`

Comment (paololim) [reply api/app/aws/EC2ContainerService.scala]: Nice, looks about right:
```
scala>       for(count <- 1 to 20){
     |         val minHealthy = BigDecimal(1) - (BigDecimal(1) / BigDecimal(count))
     |         val minHealthyInt = (minHealthy * 100).toInt
     |         println(s"instance count: $count - minHealthy: $minHealthyInt")
     |       }
instance count: 1 - minHealthy: 0
instance count: 2 - minHealthy: 50
instance count: 3 - minHealthy: 66
instance count: 4 - minHealthy: 75
instance count: 5 - minHealthy: 80
instance count: 6 - minHealthy: 83
instance count: 7 - minHealthy: 85
instance count: 8 - minHealthy: 87
instance count: 9 - minHealthy: 88
instance count: 10 - minHealthy: 90
instance count: 11 - minHealthy: 90
instance count: 12 - minHealthy: 91
instance count: 13 - minHealthy: 92
instance count: 14 - minHealthy: 92
instance count: 15 - minHealthy: 93
instance count: 16 - minHealthy: 93
instance count: 17 - minHealthy: 94
instance count: 18 - minHealthy: 94
instance count: 19 - minHealthy: 94
instance count: 20 - minHealthy: 95
```

**Comment by mbryzek** [comment]
will this make deploys take longer?

if so suggest we do a bit more aggressively

**Comment by paololim** [comment]
@mbryzek yes, this will make deploys of larger services go longer, but given load on FN - we are at a point where we may not be able to afford half of the experience or shopify or harm clusters going down, for example, and need to do it more deliberately (i.e. only 1 instance at a time)

**Comment by paololim** [comment]
Another idea would be to introduce a "min healthy instances" config where we explicitly define a number of instances that need to stay up at any given time, including during a deploy. thsi will then inform what we can set in the minhealthy cluster setting in ECS. if not defined in .delta, then will default to the formula in this pr

**Comment by mbryzek** [comment]
in the logic for a cluster with say 8 instances- i think we should do at least 2 at a time (maybe 3) vs 1 by 1

**Comment by mbryzek** [comment]
why don't we copy the settings from k8s here - will be good to socialize with team and they should just carry over to k8s.

**Comment by paololim** [comment]
@dasmb @mbryzek Where can i see the relevant k8s settings that define something similar to this?

**Comment by mbryzek** [comment]
i don't have off hand but pretty sure it is a config setting on the k8s deployment (https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)

**Pull request merged**